### DISCUSSION TITLE
How do I edit the transition duration for Chakra Drawer?

### DISCUSSION THREAD

**Discussion by sengho66** [chakra-ui/chakra-ui#4547] [Q&A]
Not sure if it’s just me but I can’t find the style props option to change the transition duration for the drawer both when open and close. How do I go about this? Thanks!!!

**Comment by jrnxf** [comment]
@segunadebayo this question is over a year old, is there any answer for this? Seems like something we could expose via a `duration` prop

**Reply by untimated** [reply]
no solution yet?

**Comment by niccmac** [comment]
Adding 
      `    transitionDuration={"100ms"}`
to DrawerContent slows down the opening of the drawer - but the exit is still super fast. I'd love to see a solution here too.

**Reply by shashwa7-dev** [reply]
Warning : NOT a solution / Possible Solution.

check this out: https://github.com/chakra-ui/chakra-ui/blob/main/packages/components/modal/stories/drawer.stories.tsx#L26

Enter transition duration works fine but still the exit  transition duration is buggy, seems like the drawer component  pops off the DOM tree before the exit transition time ends, change pos 'x' to 'y' incase your drawer is vertically positioned.

export const WithCustomMotion = () => {
  const [open, setOpen] = React.useState(false)
  return (
    <>
      <button onClick={() => setOpen(!open)}>Open</button>
      <Drawer isOpen={open} onClose={() => setOpen(false)}>
        <DrawerOverlay />
        <DrawerContent
          motionProps={{
            variants: {
              enter: {
                x: "0%",
                transition: { duration: 0.2 },
              },
              exit: {
                x: "100%",
                transition: { duration: 0.1 },
              },
            },
          }}
        >
          <div>This is the drawer content</div>
          <button>This is a button</button>
        </DrawerContent>
      </Drawer>
    </>
  )
}

Let me know if anyone has figured it out!

**Reply by YanAnderson-bit** [reply]
I tried and still not working :(

**Comment by yangdotyang** [comment]
The DrawerContent component says it takes specifications like `transitionDuration` and `transitionDelay` but, when I supply those, it only applies to the enter animation, not the exit animation. Seems like a bug!

**Comment by ismena0798dev** [comment]
Apparently this was addressed in this [PR](https://github.com/chakra-ui/chakra-ui/pull/6679), and now you should be able to pass a `motionProps` object with your transition behavior. For example for modifying the AccordionPanel transition duration:

`
<AccordionPanel
            padding="0"
            motionProps={{
              transition: {
                exit: { duration: 0.5 },
                enter: { duration: 1 },
              },
            }}
          />
`
